frappe/public/js/frappe/form/print.js

```diff
@@ -326,3 +494,88 @@ frappe.ui.get_print_settings = function (pdf, callback, letter_head) {
 		callback(data);
 	}, __("Print Settings"));
 }
+
+
+// qz tray connection wrapper
+//  - allows active and inactive connections to resolve regardless
+//  - try to connect once before firing the mimetype launcher
+//  - if connection fails, catch the reject, fire the mimetype launcher
+//  - after mimetype launcher is fired, try to connect 3 more times
+//  - display success/fail message to user
+frappe.ui.form.qz_connect = function () {
+	return new Promise(function (resolve, reject) {
+		if (qz.websocket.isActive()) { // if already active, resolve immediately
+			// frappe.show_alert({message: __('QZ Tray Connection Active!'), indicator: 'green'});
+			resolve();
+		} else {
+			// try to connect once before firing the mimetype launcher
+			frappe.show_alert({
+				message: __('Attemting Connection to QZ Tray!'),
```

Comment:
attempting (spelling)

Also no exclamation mark!

Reply:
oops... ran spell check and corrected a few other spellings as well.